coderd/workspaceagents.go

```diff
@@ -1126,7 +1126,7 @@ func convertWorkspaceAgent(derpMap *tailcfg.DERPMap, coordinator tailnet.Coordin
 		ConnectionTimeoutSeconds:     dbAgent.ConnectionTimeoutSeconds,
 		TroubleshootingURL:           troubleshootingURL,
 		LifecycleState:               codersdk.WorkspaceAgentLifecycle(dbAgent.LifecycleState),
-		LoginBeforeReady:             dbAgent.LoginBeforeReady,
```

Comment:
It is more like double-checking facts:

An old agent talking with the updated Coderd. Will the agent explode?

Reply:
Good check! It won't explode, but if, somehow, an old agent binary is started from a cache vs downloading the latest one, then the behavior could change from non-blocking to blocking. I'll add this field back to this struct and mark it deprecated.

Reply:
Fixed!